library/src/commonMain/kotlin/com/mercury/sqkon/db/QueryExt.kt

```diff
@@ -1,23 +1,47 @@
 package com.mercury.sqkon.db
 
 import app.cash.sqldelight.db.SqlPreparedStatement
+import com.mercury.sqkon.db.serialization.SqkonSerializer
 import kotlin.reflect.KProperty1
+import kotlin.reflect.KType
+import kotlin.reflect.typeOf
+
+@PublishedApi
+internal val AnyType = typeOf<Any>()
+
+// FIXME This is not ideal, but KProperty1<T, out V> doesn't have the right
+//   variance to prevent Any from satisfying V.
+/**
+ * Check that we aren't coercing types to Any at runtime.
+ */
+@PublishedApi
+internal inline fun <reified T> checkedTypeOf(): KType {
+    val type = typeOf<T>()
+    if (type == AnyType) {
+        throw IllegalArgumentException(
+            "Type was coerced to Any; did you pass the correct value?"
+        )
+    }
+    return type
+}
```

Comment:
FYI this doesn't work because we fail in `KProperty1.builder()` before hitting `checkedTypeOf`.